lib/db_utils.go

```diff
@@ -476,8 +476,37 @@ type DBPrefixes struct {
 	// 	<prefix, expirationBlockHeight, PKID, partialID> -> <>
 	PrefixNoncePKIDIndex []byte `prefix_id:"[77]" is_state:"true"`
 
-	// NEXT_TAG: 78
+	// PrefixValidatorByPKID: Retrieve a validator by PKID.
+	// Prefix, ValidatorPKID -> ValidatorEntry
+	PrefixValidatorByPKID []byte `prefix_id:"[78]" is_state:"true"`
 
+	// PrefixValidatorByStake: Retrieve the top N validators by stake.
+	// Prefix, TotalStakeAmountNanos, MaxUint64 - RegisteredAtBlockHeight, ValidatorPKID -> ValidatorPKID
+	// FIXME: @DH, should we duplicate the ValidatorPKID in the key and the value?
+	// Alternatively, we could just store and parse the ValidatorPKID from the key
+	// and store a struct{} as the value. That saves on space, but makes retrieving
+	// the ValidatorPKID from the key bytes more complex than just reading the value
+	// bytes directly since the key includes other preceding fields. Interesting
+	// trade-off. Curious your opinion.
+	PrefixValidatorByStake []byte `prefix_id:"[79]" is_state:"true"`
```

Comment:
In this case I would only have the ValidatorPKID in the key, and have the value be empty. The reason is that duplicating it like this can lead to bugs, and so we generally prefer to normalize wherever it's convenient to do so, as it seems to be here.

A case where I would say it's not worth normalizing is if you had an Entry as the value, since in that case having the value in the entry is useful in other places, and omitting it would be annoying. Since here you don't have that, I'd keep it only in the key.